File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    total = 0
    
    def dfs(n,t,n_len):
        n += 1
        nonlocal answer
        if n >= n_len:
            if t == target:
                answer += 1
            else:
                return False
        else:
            # print('t : ',t,n, numbers[n])
            dfs(n,t+numbers[n],n_len)
            dfs(n,t-numbers[n],n_len)
        
    dfs(-1,total,len(numbers))
    
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;
class Solution {
    public int solution(int[] numbers, int target) {
        l = numbers.length;
        t = target;
        answer = 0;
        dfs(-1,0,numbers);
        return answer;
    }
    public static int l, t, answer;
    
    public static void dfs(int idx, int sum, int[] numbers){
        idx +=1;
        if(idx == l){
            if (sum == t)
                answer++;
            return;
        }
        dfs(idx,sum+numbers[idx],numbers);
        dfs(idx,sum-numbers[idx],numbers);  
    }
}